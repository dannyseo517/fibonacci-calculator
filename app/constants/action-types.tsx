const REQUEST_FIBONACCI: string = 'REQUEST_FIBONACCI';

export {
    REQUEST_FIBONACCI,
}