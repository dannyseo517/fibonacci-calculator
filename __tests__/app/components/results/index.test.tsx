import React from 'react';
import ReactDOM from 'react-dom';
import FibonacciResult from 'Components/results';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import fibonacciReducer from 'Reducers/fibonacci';
import { REQUEST_FIBONACCI } from 'Constants/action-types';
import { Map } from 'immutable';

const mockStore = configureStore([thunk]);
const store = mockStore();

describe('Fibonacci results', () => {
    it('should render initial layout correctly', () => {
        const component = renderer.create(<Provider store={store}><FibonacciResult /></Provider>).toJSON();
        expect(component).toMatchSnapshot();
    });

    it('should render fibonacci results with inclusive', () => {
        const fibonacci = fibonacciReducer(Map(), { type: REQUEST_FIBONACCI, data: { value: 10, mode: 'inclusive' } });
        const newStore = mockStore({ fibonacciReducer: fibonacci });
        jest.spyOn(Date, 'now').mockImplementation(() => 1487076708000);
        const component = shallow(<Provider store={newStore}><FibonacciResult /></Provider>);
        component.setProps({ 'test': true });
        component.update();
        expect(component).toMatchSnapshot();
    });

    it('should render fibonacci results with recursion', () => {
        const fibonacci = fibonacciReducer(Map(), { type: REQUEST_FIBONACCI, data: { value: 10, mode: 'recursion' } });
        const newStore = mockStore({ fibonacciReducer: fibonacci });
        jest.spyOn(Date, 'now').mockImplementation(() => 1487076708000);
        const component = shallow(<Provider store={newStore}><FibonacciResult /></Provider>);
        component.setProps({ 'test': true });
        component.update();
        expect(component).toMatchSnapshot();
    });

    it('should call componentDidUpdate', () => {
        jest.useFakeTimers();
        const node = document.createElement('div');
        ReactDOM.render(<Provider store={store}><FibonacciResult /></Provider>, node);
        const fibonacci = fibonacciReducer(Map(), { type: REQUEST_FIBONACCI, data: { value: 10, mode: 'inclusive' } });
        const newStore = mockStore({ fibonacciReducer: fibonacci });
        ReactDOM.render(<Provider store={newStore}><FibonacciResult /></Provider>, node);
        jest.runAllTimers();
        // setInterval is inside componendDidMount
        expect(setInterval).toHaveBeenCalled();
    });

    it('should do nothing with default reducer', () => {
        const fibonacci = fibonacciReducer(Map(), { type: 'none' });
        const newStore = mockStore({ fibonacciReducer: fibonacci });
        const component = shallow(<Provider store={newStore}><FibonacciResult /></Provider>);
        expect(component).toMatchSnapshot();
    });
});