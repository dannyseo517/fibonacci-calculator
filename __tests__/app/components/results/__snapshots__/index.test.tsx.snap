// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fibonacci results should do nothing with default reducer 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeState": Object {
        "fibonacciReducer": Immutable.Map {},
      },
    }
  }
>
  <Connect(Results) />
</ContextProvider>
`;

exports[`Fibonacci results should render fibonacci results with inclusive 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeState": Object {
        "fibonacciReducer": Immutable.Map {
          "inputValue": 10,
          "fibonacciArr": Array [
            0,
            1,
            1,
            2,
            3,
            5,
            8,
            13,
            21,
            34,
          ],
        },
      },
    }
  }
>
  <Connect(Results) />
</ContextProvider>
`;

exports[`Fibonacci results should render fibonacci results with recursion 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeState": Object {
        "fibonacciReducer": Immutable.Map {
          "inputValue": 10,
          "fibonacciArr": Array [
            0,
            1,
            1,
            2,
            3,
            5,
            8,
            13,
            21,
            34,
          ],
        },
      },
    }
  }
>
  <Connect(Results) />
</ContextProvider>
`;

exports[`Fibonacci results should render initial layout correctly 1`] = `
<div
  className="results-container"
>
  <div
    className="results-heading-text"
  >
    Result (sum: 
    0
    )
  </div>
  <div
    className="results-fibonacci"
  />
</div>
`;
